### DISCUSSION TITLE
Overriding mason `ensure_installed`

### DISCUSSION THREAD

**Discussion by d3caf** [LazyVim/LazyVim#3068] [Q&A] [answered]
Hey there,

I've seen a lot of info about how to modify mason's `ensure_installed` config, but I'm struggling a bit still. I want to try using Lexical instead of elixir-ls but I don't want to remove the `lang.elixir` extra since it seems to include more than just the lsp config. Is there a way I can just remove `elixir-ls` from `ensure_installed` while still keeping the extra and also not just setting `ensure_installed = {}`?

I've made a number of attempts and elixir-ls constantly gets reinstalled no mater what I've tried. This felt like the right path, but didn't yield any results.

```lua
  {
    "williamboman/mason.nvim",
    opts = function(_, opts)
      opts.ensure_installed["elixirls"] = nil
    end,
  },
```

I've spent a lot of time reading docs for both mason and lazyvim and am not really sure what else to try. It's driving me crazy!

**Comment by dpetka2001** [comment, marked as answer]
This should do the trick
```lua
  {
    "williamboman/mason.nvim",
    opts = function(_, opts)
      opts.ensure_installed = vim.tbl_filter(function(name)
        return not vim.tbl_contains({ "elixir-ls" }, name)
      end, opts.ensure_installed)
    end,
  },
```

**Reply by d3caf** [reply]
Nice! Thanks so much. My lua illiteracy was making it hard for me to figure out how to filter out items in a table.
